confdb/aclocal_util.m4

```diff
@@ -64,7 +64,7 @@ dnl argument is already present in the variable
 AC_DEFUN([PAC_APPEND_FLAG],[
 	AC_REQUIRE([AC_PROG_FGREP])
 	AS_IF(
-		[echo "$$2" | $FGREP -e "$1" >/dev/null 2>&1],
+		[echo "$$2" | $FGREP -e "$1\b" >/dev/null 2>&1],
```

Comment:
`\b` will be treated literally since this is FGREP. See the discussion in https://github.com/pmodels/mpich/pull/5336.

I think the `-w` solution is worth trying, but we should make configure test for it first. Maybe if it doesn't work we just skip the de-dupe step.

Reply:
Hmm, this seems inconsistent with what I see on Jenkins nodes. Even with `$FGREP` the `\b` still worked and fixed the `-lcuda` problem. But, according to #5336, it should not work.

I had a separate discussion with Hui and found out that even with `-w` or `\b`, the checking would still be incorrect that `-Wl, static` can match with `-Wl`.

Do we really need this checking? I see no problem having duplicated flags.